docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,220 @@
+
+Supply keys to elements inside a `map()` function and not directly to an HTML element. Keys used within groups should be unique to each other. Keys cannot be accessed as a `prop`.
+
+**Example: Incorrect key usage:** _Should not assign a `key` in this manner._
+
+```javascript{2}
+function Number(props) {
+  return <li key={"item-" + props.value}>{props.value}</li>;
+}
+
+function NumberList(props) {
+  const numbers = props.numbers
+  return <ul>{numbers.map((item) => <Number value={item} />}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+**Example: Correct key usage:** _Assigns `key` inside `map()` function._
+
+```javascript{7}
+function Number(props) {
+  return <li>{props.value}</li>;
+}
+
+function NumberList(props) {
+  const numbers = props.numbers
+  return <ul>{numbers.map((item) => <Number key={"item-" + props.value} value={item} />)}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010)
+
+## Conditional Rendering
+
+You can create distinct components that encapsulate behavior you need. Then, you can render only the specific component you need, depending on the state of your application.
+
+### Element Variables
+
+You can use variables to store elements. This can help you render the specific element you need based on the state of your application.
+
+In the example below, we want to display a login or logout button. In `render()` we check `state.loggedIn` and assign to `button` either `<LoginButton />` or `<LogoutButton />`. We then render the element.
+
+```javascript{23,25,29}
+function LogoutButton(props) {
+  return <button onClick={props.onClick}>Logout</button>
+}
+
+function LoginButton(props) {
+  return <button onClick={props.onClick}>Login</button>
+}
+
+class LoginControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loggedIn: true};
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+  login() {
+    this.setState({loggedIn: true});
+  }
+  logout() {
+    this.setState({loggedIn: false});
+  }
+  render() {
+    let button = <LoginButton onClick={this.login} />;
+    if (this.state.loggedIn) {
+      button = <LogoutButton onClick={this.logout} />;
+    }
+    return (
+      <div>
+        {button}
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(<LoginControl />, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010)
+
+
+### Prevent Component Rendering
+
+To prevent components from rendering, return `null` or `false` from the `render()` function.
```

Comment:
Maybe start as "A component may also prevent its own rendering." 

It's not quite clear when you'd use each pattern so I'm not sure if it's worth diving into this here at all. But it's good to have an example somewhere so might as well be here. 

But I think a ternary should be explained first as it's a more common pattern. Or at least you should explain `{someCondition && <Stuff />}` first. It is **way** more common that parent decides which children to render, than a component decides to not render itself.